### ISSUE TITLE
Tests are failing for version 2.2.0

### ISSUE THREAD

**Issue by pozzolana93** [elastic/elasticsearch-dsl-py#1441]
To reproduce the error:
- Download source code of v2.2.0
- ```python setup.py test```

Some tests are failing:
- `test_integration/test_count/test_count_filter`
- `test_integration/test_document/test_save_with_tz_date`
- `test_integration/test_document/test_search_with_fields`
- `test_integration/test_index/test_index_can_be_created_with_settings_and_mappings`
- `test_integration/test_mapping/test_mapping_gets_updated_from_es`
- `test_integration/test_search/test_scan_respects_doc_types`
- `test_integration/test_search/test_scan_iterates_through_all_docs`
- `test_integration/test_search/test_multi_missing`

Some other tests were failing but due to a deprecation of fixture calls in `pytest>=4.0`, limiting the version of `pytest<4.0` and `pytest-cov<2.10` removed the errors.

**Comment by sethmlarson** [comment]
Elasticsearch v2 is EOL and no longer supported so it's unlikely that this will be fixed. Sorry about that.

**Comment by pozzolana93** [comment]
I know, but I was just wondering how is possible that some tests are failing for a released version

**Comment by pozzolana93** [comment]
Ok, I didn't know the tests under `test_integration` were actually testing the integration with the elasticsearch server (as the name suggests :) ).
I ran tests for version `0.0.11` and everything worked fine because I was running Elasticsearch 1.7.6, but when I tested version `2.2.0` I didn't switch to Elasticsearch 2.x, so the integration was failing.
Testing with Elasticsearch 2.4.6 worked fine.
Sorry for the unnecessary trouble.
